File: app/src/main/java/com/faendir/lightning_launcher/multitool/proxy/Menu.java
package com.faendir.lightning_launcher.multitool.proxy;

import android.view.View;

/**
 * @author lukas
 * @since 08.07.18
 */
public interface Menu extends Proxy {
    int MODE_ITEM_NO_EM = 12;
    int MODE_ITEM_EM = 2;

    int getMode();

    void close();

    View addMainItem(String text, Function action);
}

File: app/src/main/java/com/faendir/lightning_launcher/multitool/proxy/Menu.kt
package com.faendir.lightning_launcher.multitool.proxy

import android.view.View

/**
 * @author lukas
 * @since 08.07.18
 */
interface Menu : Proxy {

    val mode: Int

    fun close()

    fun addMainItem(text: String, action: Function): View

    companion object {
        const val MODE_ITEM_NO_EM = 12
        const val MODE_ITEM_EM = 2
    }
}
